qiskit/transpiler/passes/mapping/layout_score.py

```diff
@@ -0,0 +1,49 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A pass to evaluate how good the layout selection was.
+No CX direction is considered.
+Saves in `layout_score` the sum of the distance off for each CX.
+The lower the number, the better the selection.
+Therefore, 0 is a perfect layout selection.
+"""
+
+from qiskit.transpiler.basepasses import AnalysisPass
+
+
+class LayoutScore(AnalysisPass):
+    """
+    Saves in `layout_score` the sum of the distance off for each CX.
+    """
+    def __init__(self, coupling_map, initial_layout=None):
```

Comment:
can we remove the `initial_layout` arg and assume that layout information comes from the property set? This will be true for all normal circumstances, and doing this will simplify the code a lot.

The same is true for two other passes: `EnlargeWithAncilla` and `FullAncillaAllocation`. They have this same arg, but it is never used anywhere. They all get their layout info from the property set. So I think it should be deleted from those too.

Reply:
I'm using it for testing. We could remove it all around in a following PR?

Reply:
yeah please remove it in all 3 passes sometime

Reply:
removed in this pass in 9b8603b4. The other passes, in a differente PR